x/mongo/driver/topology/topology.go

```diff
@@ -69,6 +69,32 @@ const (
 	SingleMode
 )
 
+type hostEnv int
+
+const (
+	genuine hostEnv = iota
+	cosmosDB
+	documentDB
+)
+
+const (
+	cosmosDBMsg   = `You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb`
+	documentDBMsg = `You appear to be connected to a DocumentDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/documentdb`
+)
+
+var (
+	thirdPartySuffixes = map[string]hostEnv{
```

Comment:
Optional: If we create a separate function (`logTopologyThirdPartyUsage`), then there is no need to create package-level variables, we can nest these declarations within the logging function to give them context.